# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../nedit/nedit-5.3-makefile.diff
# Copyright (C) 2004 - 2005 The T2 SDE Project
# Copyright (C) 1998 - 2003 ROCK Linux Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

diff -urN nedit-5.3.old/makefiles/Makefile.linux nedit-5.3/makefiles/Makefile.linux
--- nedit-5.3.old/makefiles/Makefile.linux	2002-03-07 08:42:55.000000000 -0800
+++ nedit-5.3/makefiles/Makefile.linux	2003-07-25 16:13:00.000000000 -0800
@@ -5,7 +5,7 @@
 # For editres, add -DEDITRES to CFLAGS and -lXmu to LIBS
 # If using a Motif 2.1 compatible library (LessTif, OM) add
 # a '-lXp' in front of the -lXext in LIBS. You also drop the
-# -lXpm from that list.
+# from that list.
 # Ensure that the Motif/LessTif headers and libs are found!
 # e.g. you may have to add something like 
 #   -I/usr/lesstif/include  to CFLAGS and
@@ -14,8 +14,8 @@
 # To evaluate an alternative layout for the Replace/Find dialog, add
 # -DREPLACE_SCOPE to the CFLAGS. See the README file for more information.
 #
-CFLAGS=-O -I/usr/X11R6/include -DUSE_DIRENT -DUSE_LPR_PRINT_CMD
+CFLAGS=-O -I/usr/X11R6/include -D__LINUX__ -DUSE_DIRENT -DUSE_LPR_PRINT_CMD
 ARFLAGS=-urs
-LIBS= -L/usr/X11R6/lib -Wl,-Bstatic -lXm -Wl,-Bdynamic -lXp -lXpm -lXext -lXt -lSM -lICE -lX11 -lm
+LIBS= -L/usr/X11R6/lib -lXm -lXp -lXext -lXt -lSM -lICE -lX11 -lm -lXmu
 
 include Makefile.common
diff -urN nedit-5.3.old/source/preferences.c nedit-5.3/source/preferences.c
--- nedit-5.3.old/source/preferences.c	2002-05-16 00:53:09.000000000 -0700
+++ nedit-5.3/source/preferences.c	2003-01-19 17:44:57.000000000 -0800
@@ -271,14 +271,14 @@
 #ifndef VMS
 #ifdef linux
     {"shellCommands", "ShellCommands", PREF_ALLOC_STRING, "spell:Alt+B:s:EX:\n\
-	cat>spellTmp; xterm -e ispell -x spellTmp; cat spellTmp; rm spellTmp\n\
+	cat>spellTmp; xterm -e aspell -c -x spellTmp; cat spellTmp; rm spellTmp\n\
 	wc::w:ED:\nset wc=`wc`; echo $wc[1] \"lines,\" $wc[2] \"words,\" $wc[3] \"characters\"\n\
 	sort::o:EX:\nsort\nnumber lines::n:AW:\nnl -ba\nmake:Alt+Z:m:W:\nmake\n\
 	expand::p:EX:\nexpand\nunexpand::u:EX:\nunexpand\n",
     	&TempStringPrefs.shellCmds, NULL, True},
 #elif __FreeBSD__
     {"shellCommands", "ShellCommands", PREF_ALLOC_STRING, "spell:Alt+B:s:EX:\n\
-      cat>spellTmp; xterm -e ispell -x spellTmp; cat spellTmp; rm spellTmp\n\
+      cat>spellTmp; xterm -e aspell -c -x spellTmp; cat spellTmp; rm spellTmp\n\
       wc::w:ED:\nset wc=`wc`; echo $wc[1] \"words,\" $wc[2] \"lines,\" $wc[3] \"characters\"\n\
       sort::o:EX:\nsort\nnumber lines::n:AW:\npr -tn\nmake:Alt+Z:m:W:\nmake\n\
       expand::p:EX:\nexpand\nunexpand::u:EX:\nunexpand\n",
@@ -786,6 +786,8 @@
     {"shell", "Shell", PREF_STRING,
 #if defined(__MVS__) || defined(__EMX__)
     	"/bin/sh",
+#elif __LINUX__
+	"/bin/sh",
 #else
         "/bin/csh",
 #endif
